python/pyspark/worker.py

```diff
@@ -238,7 +284,8 @@ def read_udfs(pickleSer, infile, eval_type):
         # In the special case of a single UDF this will return a single result rather
         # than a tuple of results; this is the format that the JVM side expects.
         for i in range(num_udfs):
-            arg_offsets, udf = read_single_udf(pickleSer, infile, eval_type, runner_conf)
+            arg_offsets, udf = read_single_udf(
+                pickleSer, infile, eval_type, runner_conf, udf_index=i)
```

Comment:
If I understand you correctly, do you mean that we maintain a mutable list of the remaining `window_bound_type` and pass that to the `read_single_udf` function instead of using the `udf_index`?

Reply:
This isn't a big deal though, so don't block merging if the rest is ready

Reply:
@BryanCutler I don't know which way is better.. if we pass `window_eval_type` to `read_single_udf` then other non window codepath needs to handle that as well. 

Yeah I agree we can revisit this later if needed.